modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -217,84 +172,102 @@ def get_files_info(ip, _rport, shares, info)
 
     logdata = ''
 
-    list = shares.collect { |e| e[0] }
-    list.each do |x|
-      x = x.strip
-      if (x == 'ADMIN$') || (x == 'IPC$')
+    shares.each do |share|
+      share_name = share[:name].strip
+      if SKIPPABLE_SHARES.include? share_name
+        next
+      end
+
+      if (share_name == USERS_SHARE) && !datastore['SpiderProfiles']
         next
       end
 
       if !datastore['ShowFiles']
-        print_status("Spidering #{x}.")
+        print_status("Spidering #{share_name}.")
+      end
+
+      begin
+        tree = simple.client.tree_connect("\\\\#{ip}\\#{share_name}")
+      rescue RubySMB::Error::UnexpectedStatusCode => e
+        print_error "Error when trying to connect to share #{share_name} - #{e.status_code.name}"
+        print_status "Spidering #{share_name} complete."
+        next
       end
+
       subdirs = ['']
-      if (x.strip == 'C$') && datastore['SpiderProfiles']
-        subdirs = profile_options(ip, x)
+      if (DEFAULT_SHARES.include? share_name) && datastore['SpiderProfiles']
+        subdirs = profile_options(tree, share)
       end
       until subdirs.empty?
-        depth = subdirs[0].count('\\')
-        if datastore['SpiderProfiles'] && (x == 'C$')
-          if depth - 2 > datastore['MaxDepth']
+        depth = subdirs.first.count('\\')
+
+        if DEFAULT_SHARES.include? share_name
+          if datastore['SpiderProfiles']
+            if (depth - 2) > datastore['MaxDepth']
+              subdirs.shift
+              next
+            end
+          else
             subdirs.shift
             next
           end
-        elsif depth > datastore['MaxDepth']
+        end
+
+        if depth > datastore['MaxDepth']
```

Comment:
Forgive me for asking this but wouldn't you be able to end up triggering `subdirs.shift` twice in your new code  if the code block above finishes either path and `depth > datastore['MaxDepth']`? The previous code didn't seem to have this possibility so I'm confused if this a mistake in the new code.

Reply:
Fixed in latest commit 👍

Reply:
Not so sure.

Here is the old code for reference:

```
  if datastore['SpiderProfiles'] && (x == 'C$')
          if depth - 2 > datastore['MaxDepth']
            subdirs.shift
            next
          end
        elsif depth > datastore['MaxDepth']
          subdirs.shift
          next
        end
```

The old code works on the following rules:

1. If both the datastore option for SpiderProfiles is set and the directory is `C$`, then check if `depth - 2 > datastore['MaxDepth']` and if so then execute `subdirs.shift` followed by `next`.
2. Alternatively if both the datastore option for SpiderProfiles is set and the directory is `C$` but `depth > datastore['MaxDepth']`, then execute `subdirs.shift` followed by `next`.

The reason for the two checks is likely due to the possibility of number wrapping assuming the number is unsigned.

In the latest code this is instead just replaced by `if DEFAULT_SHARES.include?(share_name) && datastore['SpiderProfiles'] && (depth - 2) > datastore['MaxDepth']` which fails to account for the cases where the original value of `depth` would be greater than `datastore['MaxDepth']` by itself. For example if `depth` was 4 and `datastore['MaxDepth']` was 3, the current code would fail since 4-2=2, and this would be less than 3, even though with the original code, `depth`, which is 4, would be considered larger than `datastore['MaxDepth']`, which has a value of 3, and so the `subdirs.shift` would still be executed vs the current code where it would not be executed.

Reply:
I think you are correct. I'll test this more extensively and I'll get back to you 👍